ClientCRUD.cs

```diff
@@ -58,11 +78,88 @@ public Client Create(Client client)
             client.Cp = Convert.ToInt32(Console.ReadLine());
             Console.WriteLine("Is the LastUpdatedDate:");
             client.LastUpdatedDate = Console.ReadLine();
+            
+            bool IsDniUnique = CheckUniqueDni(Convert.ToString(client.DNI));
+            if (IsDniUnique == true) 
+            {
+                allClients.Add(client);
+                Console.WriteLine("Client was added succesfully !!");
+                File.WriteAllText(clientJsonFile, JsonConvert.SerializeObject(allClients));
+                return client;
+            }
+
+            Console.WriteLine("Client was NOT added becasue DNI is duplicated !!");
+            return client;
+        }
+
+        public void Delete(string Id)
+        {
+
+            allClients = JsonConvert.DeserializeObject<List<Client>>(clientJsonString);
 
-            allClients.Add(client);
-            Console.WriteLine("Client was added succesfully !!");
-            string clientJson = JsonConvert.SerializeObject(allClients);
-            File.WriteAllText(clientJsonFile, clientJson);
+            foreach (Client element in allClients)
+            {
+                if (Convert.ToInt32(Id).Equals(element.Id))
+                {
+                    Console.WriteLine($"Idt: {Id} deleted !!");
+                    allClients.Remove(element);
+                    string clientJson = JsonConvert.SerializeObject(allClients);
+                    File.WriteAllText(clientJsonFile, clientJson);
+                    break;
+                }
+            }
+        }
+
+        public void Update(string Id)
```

Comment:
Deberíamos recibir un parámetro del tipo `Client` (tal cual esta hecho el `Create`), buscar por el `Id` del Client recibido y actualizarlo con los datos del mismo
```suggestion
        public void Update(Client Id)
```